components/ucd/category/src/lib.rs

```diff
@@ -0,0 +1,313 @@
+
+    /// A digraphic character, with first part uppercase
+    ///
+    /// Abbreviated: Lt
+    TitlecaseLetter,
+
+    /// A modifier letter
+    ///
+    /// Abbreviated: Lm
+    ModifierLetter,
+
+    /// Other letters, including syllables and ideographs
+    ///
+    /// Abbreviated: Lo
+    OtherLetter,
+
+    /// A nonspacing combining mark (zero advance width)
+    ///
+    /// Abbreviated: Mn
+    NonspacingMark,
+
+    /// A spacing combining mark (positive advance width)
+    ///
+    /// Abbreviated: Mc
+    SpacingMark,
+
+    /// An enclosing combining mark
+    ///
+    /// Abbreviated: Me
+    EnclosingMark,
+
+    /// A decimal digit
+    ///
+    /// Abbreviated: Nd
+    DecimalNumber,
+
+    /// A letterlike numeric character
+    ///
+    /// Abbreviated: Nl
+    LetterNumber,
+
+    /// A numeric character of other type
+    ///
+    /// Abbreviated: No
+    OtherNumber,
+
+    /// A connecting punctuation mark, like a tie
+    ///
+    /// Abbreviated: Pc
+    ConnectorPunctuation,
+
+    /// A dash or hyphen punctuation mark
+    ///
+    /// Abbreviated: Pd
+    DashPunctuation,
+
+    /// An opening punctuation mark (of a pair)
+    ///
+    /// Abbreviated: Ps
+    OpenPunctuation,
+
+    /// A closing punctuation mark (of a pair)
+    ///
+    /// Abbreviated: Pe
+    ClosePunctuation,
+
+    /// An initial quotation mark
+    ///
+    /// Abbreviated: Pi
+    InitialPunctuation,
+
+    /// A final quotation mark
+    ///
+    /// Abbreviated: Pf
+    FinalPunctuation,
+
+    /// A punctuation mark of other type
+    ///
+    /// Abbreviated: Po
+    OtherPunctuation,
+
+    /// A symbol of mathematical use
+    ///
+    /// Abbreviated: Sm
+    MathSymbol,
+
+    /// A currency sign
+    ///
+    /// Abbreviated: Sc
+    CurrencySymbol,
+
+    /// A non-letterlike modifier symbol
+    ///
+    /// Abbreviated: Sk
+    ModifierSymbol,
+
+    /// A symbol of other type
+    ///
+    /// Abbreviated: So
+    OtherSymbol,
+
+    /// A space character (of various non-zero widths)
+    ///
+    /// Abbreviated: Zs
+    SpaceSeparator,
+
+    /// U+2028 LINE SEPARATOR only
+    ///
+    /// Abbreviated: Zl
+    LineSeparator,
+
+    /// U+2029 PARAGRAPH SEPARATOR only
+    ///
+    /// Abbreviated: Zp
+    ParagraphSeparator,
+
+    /// A C0 or C1 control code
+    ///
+    /// Abbreviated: Cc
+    Control,
+
+    /// A format control character
+    ///
+    /// Abbreviated: Cf
+    Format,
+
+    /// A surrogate code point
+    ///
+    /// Abbreviated: Cs
+    Surrogate,
+
+    /// A private-use character
+    ///
+    /// Abbreviated: Co
+    PrivateUse,
+
+    /// Unassigned
+    ///
+    /// Abbreviated: Cn
+    Unassigned,
+}
+
+
+use self::GeneralCategory::*;
+
+// TODO: potentially change these to match pattern matching.
+// Static array contains was easier and clearer to get the first draft up,
+// but is it optimized to the same thing?
+impl GeneralCategory {
+    /// Lu | Ll | Lt
+    ///
+    /// Abbreviated: LC
```

Comment:
`use UppercaseLetter as Lu`? In any case it's probably best to "force" use of the descriptive names.